go/pkg/client/client.go

```diff
@@ -43,54 +43,78 @@ type Client struct {
 
 	suppressTableLeakWarning bool // When true, this disables the TableHandle finalizer warning.
 
-	sessionStub
-	consoleStub
-	flightStub
-	tableStub
-	inputTableStub
+	*sessionStub
+	*consoleStub
+	*flightStub
+	*tableStub
+	*inputTableStub
 
 	appServiceClient apppb2.ApplicationServiceClient
-	ticketFact       ticketFactory
+	ticketFact       *ticketFactory
+	tokenMgr         *tokenManager
 }
 
 // NewClient starts a connection to a Deephaven server.
 //
 // The client should be closed using Close() after it is done being used.
 //
-// Keepalive messages are sent automatically by the client to the server at a regular interval (~30 seconds)
+// Keepalive messages are sent automatically by the client to the server at a regular interval
 // so that the connection remains open. The provided context is saved and used to send keepalive messages.
 //
+// host, port, and auth are used to connect to the Deephaven server.  host and port are the Deephaven server host and port.
+// auth is the authorization string used to get the first token.  Examples:
+//   - "Anonymous" is used for anonymous authentication.
+//   - "io.deephaven.authentication.psk.PskAuthenticationHandler <password>" is used for PSK authentication
+//
+// If auth is set to an empty string, DefaultAuth authentication is used.
+// To see what authentication methods are available on the Deephaven server, navigate to: http://<host>:<port>/jsapi/authentication/.
+//
 // The option arguments can be used to specify other settings for the client.
 // See the With<XYZ> methods (e.g. WithConsole) for details on what options are available.
-func NewClient(ctx context.Context, host string, port string, options ...ClientOption) (*Client, error) {
+func NewClient(ctx context.Context, host string, port string, auth string, options ...ClientOption) (client *Client, err error) {
+	defer func() {
+		if err != nil && client != nil {
+			e := client.Close()
+			log.Println("Error when closing failed client: ", e)
```

Comment:
It does not seem right to me that we unconditionally print a message here.  I would think in the common case, if we get here, `client.Close()` will actually succeed and e will be nil. Shouldn't this be

```
if e != nil {
  log.Println("blah")
}
```